### PR TITLE
windows: fix sometimes crash when FDImpl.uv() is called on stdio

### PR DESCRIPTION
**Pull request by nektro**
Closes https://github.com/oven-sh/bun/issues/13653
Closes https://github.com/oven-sh/bun/issues/13705
Closes https://github.com/oven-sh/bun/issues/13702
Closes https://github.com/oven-sh/bun/issues/11401
Closes https://github.com/oven-sh/bun/issues/13750

### CODE REVIEW

**Comment by robobun** [comment]
@nektro, your commit 7ff3680cf97d43fa786472234100e8c86fd1955b is building: [`#2704`](https://buildkite.com/bun/bun/builds/2704)

**Comment by Jarred-Sumner** [comment]
can you add a test

**Comment by Jarred-Sumner** [comment]
The test segfaulted

File: src/fd.zig [hunk 1]
```diff
@@ -170,14 +170,29 @@ pub const FDImpl = packed struct {
...
-                    \\The supplier of this FileDescriptor should call 'bun.toLibUVOwnedFD'
-                    \\or 'FDImpl.makeLibUVOwned', probably where open() was called.
-                ,
-                    .{this},
-                ),
+                .system => {
+                    const w = std.os.windows;
+
+                    const S = struct {
+                        fn is_stdio_handle(id: w.DWORD, handle: w.HANDLE) bool {
```
Comment (paperclover) [inline src/fd.zig:177]: instead of declaring a struct with a function here, why not just move the function above `pub fn uv`?

**Review by paperclover — APPROVED** [review]
unfortunate that system tagged fds are introduced somewhere for the standard io handles, but this fix makes sense.

**Review by Jarred-Sumner — CHANGES_REQUESTED** [review]
Can’t merge until the test passes on windows

**Pull request merged**